src/framework/audio/synthtypes.h

```diff
@@ -47,6 +48,52 @@ enum class SynthType {
     VSTi
 };
 
+inline std::string synthTypeToString(const SynthType& type)
+{
+    switch (type) {
+    case SynthType::Fluid: {
+        static std::string fluid("fluid");
+        return fluid;
+    }
+    case SynthType::VSTi: {
+        static std::string vsti("vsti");
+        return vsti;
+    }
+    default:
+        static std::string undefined("undefined");
+        return undefined;
+    }
+}
+
+struct SynthUri : public UriQuery
+{
+    SynthType type = SynthType::Undefined;
+
+    SynthUri() = default;
+    SynthUri(const SynthType& type)
+        : UriQuery("musescore://audio/" + synthTypeToString(type)), type(type) {}
+
+private:
+    std::string synthTypeToString(const SynthType& type)
```

Comment:
Strange, two functions `inline std::string synthTypeToString(const SynthType& type)`

Reply:
damn, no idea how I've missed this

thanks, fixed